test-data/unit/check-functions.test

```diff
@@ -3158,3 +3158,20 @@ class C(A, B):
 class D(A, B):
     def f(self, z: int) -> str: pass  # E: Method "f" is not using @override but is overriding a method in class "__main__.A"
 [typing fixtures/typing-override.pyi]
+
+[case testCallableProperty]
+from typing import Callable
+
+class something_callable:
+    def __call__(self, fn) -> str: ...
+
+def decorator(fn: Callable[..., int]) -> something_callable: ...
+
+class A:
+    @property
+    @decorator
+    def f(self) -> int: ...
```

Comment:
People recently complained (don't remember where) that this use case:
```python
def make_method() -> SomeCallbackProtocol: ...
class A:
    meth = make_method()
```
is not handled the same way as
```python
def make_method_callable() -> Callable[[int], int]: ...
class B:
    meth = make_method_callable()
```
(because certain special-casing only applies to callable types). I am 95% sure this PR should fix that as well (or maybe with very few modifications), if yes, could you please add such test as well?

Reply:
Thanks for mentioning this! It's not fixed by this change, but should be an easy fix. I will make a follow up PR

Reply:
https://github.com/python/mypy/pull/15993